lib/connect-backbone-to-react.js

```diff
@@ -110,27 +99,34 @@ module.exports = function connectBackboneToReact(
         // The only case where this flag is encountered is when this component
         // is unmounted within an event handler but the 'all' event is still triggered.
         // It is covered in a test case.
-        if (this.hasBeenUnmounted) {
-          return;
-        }
+        if (this.hasBeenUnmounted) return;
 
         this.setState(mapModelsToProps(this.models, this.props));
       }
 
-      componentWillReceiveProps(nextProps, nextContext) {
-        this.setModels(nextProps, nextContext);
-        this.createNewProps();
-
+      renderChild({ models } = {}) {
+        const newModels = Object.assign({}, models, this.props.models, this.models);
         // Bind event listeners for each model that changed.
-        Object.keys(this.models).forEach(mapKey => {
-          const model = this.models[mapKey];
-          if ((this.props.models && this.props.models[mapKey] === this.models[mapKey]) ||
-            (this.context.models && this.context.models[mapKey] === this.models[mapKey])) {
+        Object.keys(newModels).forEach(mapKey => {
+          const model = newModels[mapKey];
+          if (this.models && this.models[mapKey] === newModels[mapKey]) {
             return; // Did not change.
           }
 
           this.createEventListener(mapKey, model);
+          this.models[mapKey] = model;
         });
+        validateModelTypes(this.models);
```

Comment:
We should be validating models before we create event listeners. If we don't have a test for this already we should add one.